src/stdune/path.ml

```diff
@@ -172,6 +173,11 @@ end = struct
 
   let compare_val x y = String.compare (to_string x) (to_string y)
 
+  let equal x y =
+    match compare x y with
+    | Eq -> true
+    | Gt | Lt -> false
+
```

Comment:
What I think would be nicer is to have a `Compare.Make` functor that would generate `equal`, `>=`, etc. from `compare`.